addons/binding/org.openhab.binding.solaredge/src/main/java/org/openhab/binding/solaredge/internal/callback/AbstractCommandCallback.java

```diff
@@ -0,0 +1,183 @@
+ */
+package org.openhab.binding.solaredge.internal.callback;
+
+import static org.openhab.binding.solaredge.SolarEdgeBindingConstants.*;
+
+import java.net.CookieStore;
+import java.net.HttpCookie;
+import java.net.SocketTimeoutException;
+import java.net.URI;
+import java.net.UnknownHostException;
+import java.nio.ByteBuffer;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.api.Response;
+import org.eclipse.jetty.client.util.BufferingResponseListener;
+import org.eclipse.jetty.http.HttpStatus;
+import org.eclipse.jetty.http.HttpStatus.Code;
+import org.openhab.binding.solaredge.config.SolarEdgeConfiguration;
+import org.openhab.binding.solaredge.internal.command.SolarEdgeCommand;
+import org.openhab.binding.solaredge.internal.connector.CommunicationStatus;
+import org.openhab.binding.solaredge.internal.connector.StatusUpdateListener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * base class for all commands. common logic should be implemented here
+ *
+ * @author Alexander Friese - initial contribution
+ */
+public abstract class AbstractCommandCallback extends BufferingResponseListener implements SolarEdgeCommand {
+
+    /**
+     * logger
+     */
+    protected final Logger logger = LoggerFactory.getLogger(AbstractCommandCallback.class);
+
+    /**
+     * the configuration
+     */
+    protected final SolarEdgeConfiguration config;
+
+    /**
+     * JSON deserializer
+     */
+    protected final Gson gson;
+
+    /**
+     * status code of fulfilled request
+     */
+    private CommunicationStatus communicationStatus;
+
+    /**
+     * listener to provide updates to the WebInterface class
+     */
+    private StatusUpdateListener listener;
+
+    /**
+     * the constructor
+     *
+     * @param config
+     */
+    public AbstractCommandCallback(SolarEdgeConfiguration config) {
+        this.communicationStatus = new CommunicationStatus();
+        this.config = config;
+        this.gson = new Gson();
+
+    }
+
+    /**
+     * the constructor
+     *
+     * @param config
+     */
+    public AbstractCommandCallback(SolarEdgeConfiguration config, StatusUpdateListener listener) {
+        this(config);
+        this.listener = listener;
+    }
+
+    /**
+     * Log request success
+     */
+    @Override
+    public final void onSuccess(Response response) {
+        super.onSuccess(response);
+        communicationStatus.setHttpCode(HttpStatus.getCode(response.getStatus()));
+        logger.debug("HTTP response {}", response.getStatus());
+    }
+
+    /**
+     * Log request failure
+     */
+    @Override
+    public final void onFailure(Response response, Throwable failure) {
+        super.onFailure(response, failure);
+        logger.warn("Request failed: {}", failure.toString());
```

Comment:
Again, please only log warnings if you want people to report bugs (and then better include the stacktrace). Do NOT log anything, if there is a network issue, see also https://www.openhab.org/docs/developer/development/guidelines.html#e-logging.

Reply:
changed it to debug.